addons/resource/tests/test_resource_calendar.py

```diff
@@ -65,3 +67,51 @@ def test_flexible_calendar_attendance_interval_duration(self):
         actual_duration = end - start
 
         self.assertEqual(actual_duration.seconds / 3600, calendar.full_time_required_hours, "For a full day, the interval must match full time required hours")
+
+    @freeze_time("2019-5-28 08:00:00")
+    def test_working_time_holiday_multicompany(self):
+        """
+        This test checks that there is no issue computing the "working time to assign" even if a holiday has been set
+        for this moment, but on another company.
+        """
+        company_0, company_1 = self.env['res.company'].create([{
+            "name": "Test company 0",
+        },
+            {
+                "name": "Test company 1",
+            }])
+
+        leave_1, leave_2 = self.env['resource.calendar.leaves'].create([{
+            'name': "Public Holiday for company 0",
+            'calendar_id': company_1.resource_calendar_ids.id,
+            'company_id': company_1.id,
+            'date_from': datetime(2019, 5, 27, 0, 0, 0),
+            'date_to': datetime(2019, 5, 27, 23, 0, 0),
+            'resource_id': False,
+            'time_type': "leave",
+        }, {
+            'name': "Public Holiday without company",
+            'calendar_id': False,
+            'company_id': False,
+            'date_from': datetime(2019, 6, 29, 0, 0, 0),
+            'date_to': datetime(2019, 6, 29, 23, 59, 59),
+            'resource_id': False,
+            'time_type': "leave",
+        }])
+        duration_company_0 = company_0.resource_calendar_ids.get_work_duration_data(
+            datetime(2019, 5, 27, 0, 0, 0),
+            datetime(2019, 5, 29, 23, 59, 59),
+            compute_leaves=True)
+        self.assertEqual(
+            duration_company_0,
+            {'days': 3.0, 'hours': 24.0},
```

Comment:
the test is wrong as we should have 2 days here, and only 1 for company_1 :thinking: